flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2794,6 +2799,39 @@ static mlir::Value createExtremumCompare(mlir::Location loc,
   return result;
 }
 
+// UNPACK
+fir::ExtendedValue
+IntrinsicLibrary::genUnpack(mlir::Type resultType,
+                            llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 3);
+
+  // Handle required vector argument
+  auto vector = builder.createBox(loc, args[0]);
+  fir::BoxValue vectorTmp = builder.createBox(loc, args[0]);
+  auto vectorRank = vectorTmp.rank();
+  assert(vectorRank == 1);
```

Comment:
Shouldn't the rank of the vector argument be checked in semantics?

Reply:
You're correct.  I'll remove this